#https://www.biostars.org/p/343138/
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")

BiocManager::install("biomaRt")
install.packages("BiocManager")
BiocManager::install("DESeq2")

library(biomaRt)
library(DESeq2)


sampleFiles <- NULL
sampleNames <- NULL
sampleCondition <- NULL
sampleTable <- NULL
ddsHTSeq <- NULL

#Reading count files that were generated by HTSeq count
directory <- "C:/Users/Karni Bedirian/Desktop/DESeq_analysis/pipeline/TILs_counts"
sampleFiles <- grep("counts",list.files(directory),value=TRUE) #reading the count files
sampleNames <- sampleFiles
sampleNames <-   gsub("Aligned.out.sorted.counts", "", sampleNames)

list <- read.csv("ConditionVsError_top100genes.csv")

#sampleCondition <- c("CrePos","CreNeg", "CreNeg", "CreNeg", "CrePos",
     #                "CrePos", "CrePos", "CrePos", "CreNeg", "CreNeg", "CreNeg", "CreNeg",
     #                "CrePos", "CrePos", "CreNeg", "CreNeg", "CrePos", "CrePos",
      #               "CreNeg", "CreNeg", "CreNeg", "CreNeg","CreNeg", "CrePos",
      #               "CrePos", "CrePos", "CrePos", "CreNeg"  )
sampleCondition <- read.csv("Conditions.csv")
sampleCondition <- sampleCondition[1:nrow(sampleCondition),1]
sampleCondition <- as.character(sampleCondition)


sampleTable <- data.frame(sampleName = sampleNames,
                          fileName = sampleFiles,
                          condition = sampleCondition)

#creating DESeq dataset
ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                       directory = directory,
                                       design= ~ condition)

#remove those genes which have a total count of zero:
ddsHTSeq  <- ddsHTSeq [ rowSums(assay(ddsHTSeq )) >= 5, ]

#run the differential expression pipeline on the raw counts with a single call to the function DESeq:
ddsHTSeq <- DESeq(ddsHTSeq, minReplicatesForReplace=Inf) 

#Building the results table
res <- results(ddsHTSeq , cooksCutoff=FALSE, independentFiltering=FALSE)
#res <- results(ddsHTSeq)

#writing the results into a csv file
write.csv(as.data.frame(res), file = "DESeq_results_perm.csv")


